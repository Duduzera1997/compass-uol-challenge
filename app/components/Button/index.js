import React from 'react';
import PropTypes from 'prop-types';
import * as SC from './styles';

function Button({ onPress }) {
  return (
    <SC.Container>
      <SC.TouchableButton onPress={onPress} testID="component-button-press">
        <SC.Label>Visualizar Not√≠cia Completa</SC.Label>
      </SC.TouchableButton>
    </SC.Container>
  );
}

Button.propTypes = {
  onPress: PropTypes.func.isRequired,
};

export default Button;
