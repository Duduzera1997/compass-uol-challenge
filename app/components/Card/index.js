import React from 'react';
import PropTypes from 'prop-types';
import { format } from 'date-fns';
import ptBR from 'date-fns/locale/pt-BR';
import * as SC from './styles';

function Card({ imageUrl, title, abstract, noticeDate }) {
  return (
    <SC.ContainerCard>
      <SC.CardTouchable>
        <SC.CardContent flex={1}>
          <SC.PreviewImage source={{ uri: imageUrl }} />
        </SC.CardContent>
        <SC.CardContent flex={2.5}>
          <SC.Text size={15} fontBold>
            {title}
          </SC.Text>
          <SC.TextDate size={11.5} fontBold>
            {format(new Date(noticeDate), "dd 'de' MMMM 'de' yyyy", {
              locale: ptBR,
            })}
          </SC.TextDate>
          <SC.Text numberOfLines={2} size={13}>
            {abstract}
          </SC.Text>
        </SC.CardContent>
      </SC.CardTouchable>
    </SC.ContainerCard>
  );
}

Card.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  abstract: PropTypes.string.isRequired,
  noticeDate: PropTypes.string.isRequired,
};

export default Card;
