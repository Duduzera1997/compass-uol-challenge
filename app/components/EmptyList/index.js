import React from 'react';
import PropTypes from 'prop-types';
import * as SC from './styles';

function EmptyList({ isLoading }) {
  return (
    <SC.ContainerEmptyList>
      {isLoading && <SC.ActivityIndicator />}
      <SC.Label>
        {isLoading
          ? 'Buscando...'
          : 'Ooops! ðŸ˜ž\n Nada foi encontrado, tente novamente...'}
      </SC.Label>
    </SC.ContainerEmptyList>
  );
}

EmptyList.defaultProps = {
  isLoading: false,
};

EmptyList.propTypes = {
  isLoading: PropTypes.bool,
};

export default EmptyList;
