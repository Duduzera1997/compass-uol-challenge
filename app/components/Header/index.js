import PropTypes from 'prop-types';
import React from 'react';

import * as SC from './styles';
import { formatDateLiteral } from '~/utils';

function Header({ headerText, dateText }) {
  return (
    <SC.ContainerHeader>
      <SC.HeaderTitle>{headerText} News</SC.HeaderTitle>
      <SC.DateLabel>Hoje Ã©, {dateText}.</SC.DateLabel>
    </SC.ContainerHeader>
  );
}

Header.defaultProps = {
  dateText: formatDateLiteral(new Date()),
};

Header.propTypes = {
  headerText: PropTypes.string.isRequired,
  dateText: PropTypes.string,
};

export default Header;
