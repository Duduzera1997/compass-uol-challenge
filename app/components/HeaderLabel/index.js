import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';

function HeaderLabel({ label }) {
  return (
    <View>
      <Text>{label} News</Text>
    </View>
  );
}

HeaderLabel.defaultProps = {
  label: 'Best',
};

HeaderLabel.propTypes = {
  label: PropTypes.string,
};

export default HeaderLabel;
