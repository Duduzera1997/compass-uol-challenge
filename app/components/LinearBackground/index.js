import LinearGradiend from 'react-native-linear-gradient';
import styled from 'styled-components/native';

import { darkPurple, red } from '~/styles/colors';

export default styled(LinearGradiend).attrs({
  colors: [darkPurple, red],
  start: { x: 0, y: 0 },
  end: { x: 0, y: 1.05 },
})`
  flex: 1;
`;
