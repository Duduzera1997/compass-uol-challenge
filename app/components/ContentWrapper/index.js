import React from 'react';
import PropTypes from 'prop-types';
import * as SC from './styles';

import Header from '../Header';
import LinearBackground from '../LinearBackground';
import SearchInput from '../SearchInput';
import ViewArticleModal from '../ViewArticleModal';

function ContentWrapper({
  title,
  children,
  showSearchInput,
  onSeachInputTextChange,
  modalProps: { visible, item, onRequestClose },
}) {
  return (
    <LinearBackground>
      <SC.Container>
        <ViewArticleModal
          visible={visible}
          item={item}
          onRequestClose={onRequestClose}
        />
        <SC.ContainerByFlex flex={1}>
          <SC.ContainerByFlex flex={1}>
            <Header headerText={title} />
          </SC.ContainerByFlex>
          {showSearchInput && (
            <SC.ContainerByFlex flex={1}>
              <SearchInput onChangeText={onSeachInputTextChange} />
            </SC.ContainerByFlex>
          )}
          <SC.ContainerByFlex flex={6}>{children}</SC.ContainerByFlex>
        </SC.ContainerByFlex>
      </SC.Container>
    </LinearBackground>
  );
}

ContentWrapper.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  showSearchInput: PropTypes.bool.isRequired,
  onSeachInputTextChange: PropTypes.func.isRequired,
  modalProps: PropTypes.shape({
    visible: PropTypes.bool.isRequired,
    item: PropTypes.shape({}),
    onRequestClose: PropTypes.func,
  }).isRequired,
};

export default ContentWrapper;
