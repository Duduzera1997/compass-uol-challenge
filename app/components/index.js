import TabIcon from './TabIcon';
import LinearBackground from './LinearBackground';
import Loading from './Loading';
import Header from './Header';
import Card from './Card';
import ListHelper from './ListHelper';
import ListArticles from './ListArticles';
import ViewArticleModal from './ViewArticleModal';
import Button from './Button';
import SearchInput from './SearchInput';
import ContentWrapper from './ContentWrapper';

export {
  TabIcon,
  LinearBackground,
  Loading,
  Header,
  Card,
  ListHelper,
  ListArticles,
  ViewArticleModal,
  Button,
  SearchInput,
  ContentWrapper,
};
