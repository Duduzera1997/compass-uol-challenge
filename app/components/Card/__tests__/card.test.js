import React from 'react';
import { format } from 'date-fns';
import ptBR from 'date-fns/locale/pt-BR';

import { renderWithTheme } from '~/utils/helperTest';
import { Card } from '~/components';

describe('Card Component', () => {
  it('should be render a Card with delivered props', () => {
    const imageUrl =
      'https://upload.wikimedia.org/wikipedia/commons/b/be/Mapa_do_Brasil_com_a_Bandeira_Nacional.png';
    const title =
      'Brazil wins covid-19 and peoples return to their normal lives';
    const abstract =
      'Brazil finally managed to win Covid-19, it was the 12th country to win the virus';
    const noticeDate = new Date(2022, 6, 15).toISOString();
    const { getByText, toJSON } = renderWithTheme(
      <Card
        imageUrl={imageUrl}
        title={title}
        abstract={abstract}
        noticeDate={noticeDate}
      />,
    );
    const formatedDate = format(
      new Date(noticeDate),
      "dd 'de' MMMM 'de' yyyy",
      {
        locale: ptBR,
      },
    );
    expect(getByText(/Brazil wins covid-19/)).toBeTruthy();
    expect(getByText(/12th country to win the virus/)).toBeTruthy();
    expect(getByText(formatedDate)).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
