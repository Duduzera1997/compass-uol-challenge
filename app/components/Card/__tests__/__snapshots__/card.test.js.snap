// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should be render a Card with delivered props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 120,
        "paddingBottom": 10,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": "100%",
        "opacity": 1,
        "width": "92%",
      }
    }
  >
    <View
      flex={1}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "flex-start",
            "paddingBottom": 2,
            "paddingLeft": 7,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://upload.wikimedia.org/wikipedia/commons/b/be/Mapa_do_Brasil_com_a_Bandeira_Nacional.png",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 10,
              "height": "95%",
              "width": "100%",
            },
          ]
        }
      />
    </View>
    <View
      flex={2.5}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 2.5,
            "flexShrink": 1,
            "justifyContent": "flex-start",
            "paddingBottom": 2,
            "paddingLeft": 7,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <Text
        fontBold={true}
        size={15}
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "PTSansNarrowBold",
              "fontSize": 15,
              "paddingBottom": 2,
              "paddingLeft": 4,
              "paddingRight": 2,
              "paddingTop": 4,
              "width": "99%",
            },
          ]
        }
      >
        Brazil wins covid-19 and peoples return to their normal lives
      </Text>
      <Text
        fontBold={true}
        size={11.5}
        style={
          Array [
            Object {
              "color": "#767676",
              "fontFamily": "PTSansNarrowBold",
              "fontSize": 11.5,
              "paddingBottom": 2,
              "paddingLeft": 4,
              "paddingRight": 2,
              "paddingTop": 0,
              "width": "99%",
            },
          ]
        }
      >
        15 de julho de 2022
      </Text>
      <Text
        numberOfLines={2}
        size={13}
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "PTSansNarrowRegular",
              "fontSize": 13,
              "paddingBottom": 2,
              "paddingLeft": 4,
              "paddingRight": 2,
              "paddingTop": 4,
              "width": "99%",
            },
          ]
        }
      >
        Brazil finally managed to win Covid-19, it was the 12th country to win the virus
      </Text>
    </View>
  </View>
</View>
`;
